gpt_index/indices/base.py

```diff
@@ -145,6 +147,7 @@ def delete(self, document: BaseDocument) -> None:
     def _mode_to_query(self, mode: str, **query_kwargs: Any) -> BaseGPTIndexQuery:
         """Query mode to class."""
 
+    @llm_token_counter("query")
```

Comment:
if you put this on superclass do you still need to put on subclass?

Reply:
yes, i believe so
 
if a subclass overrides this `query()` method, then the effect of this decorator goes away. you would have to decorate that overridden method too. this makes it a bit messy :( if you can think of a better way to get around this definitely lmk!

Reply:
talked offline about a better way, described in this issue: https://github.com/jerryjliu/gpt_index/issues/101